transcom/mymove
# [MB-16382] Adds an authz check when creating a pro gear weight ticket

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16382)

## Summary

This pull request adds an authorization check in the server-side handler for the endpoint used when a customer creates a pro gear weight ticket, verifying that the user who is creating the weight ticket is the the same one that's attached (through the Orders <- Move <- Shipment) to the PPMShipment that's getting the weight ticket. Tests are added for this functionality.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Verify that the server tests pass.
2. Log in to the MyMove application as a customer user and create a PPM shipment.
3. Log in to the office application as a service counselor and approve the shipment.
4. Log back into the MyMove application as the customer from step 2, and create a weight ticket. Note the ID of the PPM shipment (it'll display as the UUID fragment in something like `/ppm-shipments/f45e42cd-a136-491d-bc8d-9f7c46089728`)
6. Using Postman/cURL, authenticate as a _different_ customer user, then make a POST request to `/ppm-shipments/{:ppmShipmentId}/pro-gear-weight-tickets` to create another weight ticket for the customer from steps 2 and 4.
7. Verify that the response is a 403 status code and that the weight ticket is not created for the other user's shipment.

### Backend
